### PR TITLE
Seacas sept snapshot

### PR DESCRIPTION
**Pull request by gsjaardema**
@trilinos/seacas 

## Description

    * Primarily spelling fixes caught by codespell
    * Update fmt:: library
    * Add comma-separated output to explore summary
    * Add some C++11 =default and =delete
    * Remove some leftover `std::numpunct` usages
    * Fix array overrun in exodus `ex_put_init_ext` name output
    * Clean up exodus format logic to unify serial and parallel
    * Skinner now skins block blocks instead of entire model [experimental]
    * Add `format_id_list` function which gathers ranges:
      -- 1-10, 13-15, 17, 18 instead of 1,2,3,4,5,6,7,8,9,10,13,14,15,17,18
      -- range and sequence separators can be specified
      -- Add use of this in a few error/warning message outputs.
    * Add Utst_utils.C to test some functions in Ioss::Utils class (`number_width` and `format_id_list`)
    * Add `Ioss::ParallelUtils::barrier()` to eliminate some `#ifdef SEACAS_HAVE_MPI` uses in code
    * Refactor Ioss::Property API to make safer to use. Eliminated defaulted argument and a never-used function
    * Better high-water memory and normal memory use routines.
    * Doxygen comment cleanup
    * Make field length optional -- can be obtained from GE that field is added to.
    * Fix size of embedded NodeBlock in an empty StructuredBlock: 0 not 1
    * Use RAII for MPI_Finalize
    * Improved `io_info` output for parallel runs.
    * Fix version handling in exodus-*.py.in
## Checklist

- [x] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] No new compiler warnings were introduced.
- [x] These changes break backwards compatibility.

### CODE REVIEW

**Comment by gsjaardema** [comment]
Sorry for the large amount of changes -- most of them are spelling fixes, but that doesn't make it any easier to review...

File: packages/seacas/applications/blot/cnvmax.f [hunk 1]
```diff
@@ -48,7 +48,7 @@ SUBROUTINE CNVMAX (NUMNPF, VARNP, IN2ELB, NMIN, NMAX, FMIN, FMAX)
 C   --   Written by Amy Gilkey - revised 03/09/88
```
Comment (mhoemmen) [inline packages/seacas/applications/blot/cnvmax.f]: Let no one forget that our software has a long lifetime :-)

File: packages/seacas/applications/conjoin/CJ_ExodusFile.h [hunk 2]
```diff
@@ -43,6 +43,8 @@ namespace Excn {
   public:
     explicit ExodusFile(size_t which);
     ~ExodusFile();
+    ExodusFile(const ExodusFile &) = delete;
+    ExodusFile operator=(const ExodusFile &) = delete;
```
Comment (mhoemmen) [inline packages/seacas/applications/conjoin/CJ_ExodusFile.h]: The usual idiom is to disable move construction and assignment as well.

File: packages/seacas/libraries/aprepro_lib/apr_builtin.cc [hunk 3]
```diff
@@ -179,7 +179,7 @@ namespace SEAMS {
     return (temp);
   }
 
-  // DO_HYPOT: calcluate sqrt(p^2 + q^2)
+  // DO_HYPOT: calculate sqrt(p^2 + q^2)
```
Comment (mhoemmen) [inline packages/seacas/libraries/aprepro_lib/apr_builtin.cc]: C++ has `std::hypot` now.

File: packages/seacas/libraries/ioss/src/Ioss_Utils.C [hunk 4]
```diff
@@ -41,9 +41,9 @@
 #include <cstdlib>
 #include <cstring>
 #include <ctime>
+#include <fmt/chrono.h>
```
Comment (mhoemmen) [inline packages/seacas/libraries/ioss/src/Ioss_Utils.C]: Are you including this for `std::chrono`?  That should live in `<chrono>`.

Comment (gsjaardema) [reply packages/seacas/libraries/ioss/src/Ioss_Utils.C]: No, this is from lib::fmt which is the prototype for the new pythonesque output library which will be included in C++20.

File: packages/seacas/libraries/aprepro_lib/apr_builtin.cc (hunk 3 above)
Comment (gsjaardema) [reply packages/seacas/libraries/aprepro_lib/apr_builtin.cc]: That's true... I will look into what can be replaced with C++11 math/numeric functions.

File: packages/seacas/applications/blot/cnvmax.f (hunk 1 above)
Comment (gsjaardema) [reply packages/seacas/applications/blot/cnvmax.f]: Yes, older than some of the people running it...

File: packages/seacas/applications/conjoin/CJ_ExodusFile.h (hunk 2 above)
Comment (gsjaardema) [reply packages/seacas/applications/conjoin/CJ_ExodusFile.h]: Yes, I should add those everywhere...

**Comment by gsjaardema** [comment]
There is a non-backward compatible change in this set -- Removed a defaulted parameter from the Ioss::Property constructor.  Stk_IO was using this parameter (at its defaulted value) so I removed those uses.

**Comment by gsjaardema** [comment]
@alanw0 You might want to be aware of commit ad74712 that I included with this PR.  My PR removed the default parameter in the Ioss::Property constructor that defaulted to false.  The IossBridge file was using this parameter a few places (always with false) so I removed those. The parameter can never be `true`, so there was no reason for the parameter to exist.

**Comment by alanw0** [comment]
Thanks Greg, I'll go ahead and remove those 'false' usages from sierra's copy of IossBridge.C

**Comment by gsjaardema** [comment]
@mhoemmen Can you review one last time -- all tests and builds are now passing.  Thanks.

**Comment by gsjaardema** [comment]
@mhoemmen Thanks.  For some reason it wasn't showing up till I sent the message...  Sorry for the unneeded message

**Pull request merged**